src/SaveHandler/DbTableGateway.php

```diff
@@ -129,9 +129,9 @@ public function write($id, $data)
         $rows = $this->tableGateway->select([
             $this->options->getIdColumn()   => $id,
             $this->options->getNameColumn() => $this->sessionName,
-        ]);
+        ])->current();
```

Comment:
Changes like these require a failing test case validating that the bug was present and fixed afterwards.